### PR TITLE
Fixed flask template merging issues by using the composition order param

### LINKED ISSUE #1148: Getting error creating app with Master Detail page
Im using Ubuntu18.04 all packages updated.

> node10.16.3
> Templates version: 0.0.19229.1
> Wizard version: 0.0.1922901

Vue.js
Flask
when i create and app and select Master Detail page it fails saying Templates could not be generated.

Screenshot from 2019-08-23 23-02-40

### PR DESCRIPTION
**Pull request by SahilTara**
Summary of changes:
- Reverted name "TheFlask" to Flask for grid.
- Used compositionOrder for all problematic flask templates instead of messing around with the alphabetical structure of items (i.e. changing the names to things like "TheFlask"), much cleaner and no longer feels like a hack.

closes #1148

### CODE REVIEW

File: templates/Web/_composition/Flask/Page.Flask.Grid.AddMerging/.template.config/template.json [hunk 1]
```diff
@@ -3,13 +3,14 @@
   "classifications": [
     "Universal"
   ],
-  "name": "Page.TheFlask.Grid.AddMerging",
+  "name": "Page.Flask.Grid.AddMerging",
```
Comment (Tanya0609) [inline templates/Web/_composition/Flask/Page.Flask.Grid.AddMerging/.template.config/template.json:6]: nice! we don't depend on the tricky solution now!

**Review by Tanya0609 — APPROVED** [review]
Great work! Tested on Mac, to generate a Vue Flask with Grid page, it still works!

File: templates/Web/_composition/Flask/Page.Flask.AddEndPoints/.template.config/template.json [hunk 2]
```diff
@@ -10,6 +10,7 @@
     "wts.type": "composition",
     "wts.platform": "Web",
     "wts.version": "1.0.0",
+    "wts.compositionOrder": "0",
```
Comment (jasmingoh) [inline templates/Web/_composition/Flask/Page.Flask.AddEndPoints/.template.config/template.json:13]: just out of curiosity, where is `compositionOrder` used?

Comment (SahilTara) [reply templates/Web/_composition/Flask/Page.Flask.AddEndPoints/.template.config/template.json:13]: It's used in core in the following areas:
https://github.com/microsoft/CoreTemplateStudio/blob/e97c1c579b2c0719b1abaaf30e2023a3a0089d16/code/src/CoreTemplateStudio/CoreTemplateStudio.Core/Templates/ITemplateInfoExtensions.cs#L298-L311

https://github.com/microsoft/CoreTemplateStudio/blob/e97c1c579b2c0719b1abaaf30e2023a3a0089d16/code/src/CoreTemplateStudio/CoreTemplateStudio.Core/Gen/GenComposer.cs#L148-L191

**Review by smmatte — APPROVED** [review]

**Review by jcoc611-microsoft — APPROVED** [review]
I'm guessing this is supported with the current CoreTS version in this repo? If not we should update it as well

**Comment by SahilTara** [comment]
> I'm guessing this is supported with the current CoreTS version in this repo? If not we should update it as well

It has actually been supported for a couple months. Just hasn't needed to be used too much until flask was introduced apparently.

**Pull request merged**